### PR TITLE
chore: some minor cleanups

### PR DESCRIPTION
**Pull request by henryiii**
This makes a few minor fixes detected and while looking at the code and via
some automated tooling. If this looks good, I'll probably have a few followups,
ending with a pre-commit file that will help keep this clean and tidy. :)

- docs: fix spelling
- chore: clean up some minor stylistic issues
- fix: always include build-backend when using pyproject.toml

### CODE REVIEW

File: tests/test_classes.py [hunk 1]
```diff
@@ -129,7 +129,8 @@ def test06_reference_internal(clean):
 
 def test07_big():
     x = [t.Big() for i in range(1024)]
-    x = [t.BigAligned() for i in range(1024)]
+    x2 = [t.BigAligned() for i in range(1024)]
+    assert x == x2
```
Comment (henryiii) [inline tests/test_classes.py:133]: FYI, not sure what was intended here. Guessed maybe this?

**Comment by wjakob** [comment]
These all look fine to me, except for the equality comparison thingy. I am not used to the pre-commit workflow, and there is a chance that I won't like it. I am mentioning this in case it's a lot of work for you to set up, in which case that work might be wasted.

Things that I don't want to do that in this project are:

- formatting everything with clang-format (it's fine to provide a .clang-format file for new people contributing code, but I don't want automation for this)
- adding lots of comments throughout the codebase to silence warnings produced by clang-tidy.

**Comment by wjakob** [comment]
(I'm confused why the CI did not run for this PR. I had pressed a button to give permission, but then nothing seems to have happened.)

**Comment by henryiii** [comment]
(GHA is out for a bit - https://www.githubstatus.com)

Not setting up clang-tidy (and that's not something you can put in pre-commit anyway). I'll avoid clang-format (though it works beautifully in pre-commit these days).

File: tests/test_classes.py (hunk 1 above)
Comment (henryiii) [reply tests/test_classes.py:133]: Apparently that is not what was intended here. Is this just supposed to run and not check anything?

**Comment by Skylion007** [comment]
@wjakob Happy to setup pre-commit workflow. It's not a lot of work at all.

File: tests/test_holders.py [hunk 2]
```diff
@@ -139,7 +139,7 @@ def test07_uniqueptr_passthrough(clean):
     t.reset()
 
     with pytest.warns(RuntimeWarning, match=r'nanobind::detail::nb_relinquish_ownership()'):
-        with pytest.raises(TypeError) as excinfo:
+        with pytest.raises(TypeError):
             assert t.passthrough_unique(t.Example(1)).value == 1
```
Comment (Skylion007) [inline tests/test_holders.py:143]:
```suggestion
with pytest.warns(RuntimeWarning, match=r'nanobind::detail::nb_relinquish_ownership()'), pytest.raises(TypeError):
        assert t.passthrough_unique(t.Example(1)).value == 1
```

Comment (henryiii) [reply tests/test_holders.py:143]: You have to make the suggestion on the new line, not the deleted line, for it to be automatically apply-able.

**Comment by henryiii** [comment]
I've already got the pre-commit file ready and will add it in a PR or two, but I think @wjakob meant running locally on his machine.

That's also really easy, and very worthwhile, IMO.

You can also set up pre-commit.ci and have it automatically run on all PRs, which is great (once there's a file here).

File: tests/test_holders.py (hunk 2 above)
Comment (Skylion007) [inline tests/test_holders.py:143]:
```suggestion
with pytest.warns(RuntimeWarning, match=r'nanobind::detail::nb_relinquish_ownership()'), pytest.raises(TypeError):
        assert t.passthrough_unique(t.Example(1)).value == 1
```

File: tests/test_classes.py (hunk 1 above)
Comment (henryiii) [reply tests/test_classes.py:133]:
```suggestion
```

**Review by wjakob — COMMENTED** [review]
I left some comments but then never submitted them, sorry.

**Comment by wjakob** [comment]
I will merge this -- still curious about the `build-backend` change.

File: pyproject.toml [hunk 3]
```diff
@@ -1,5 +1,7 @@
 [build-system]
 requires = ["setuptools>=42", "wheel", "scikit-build", "cmake>=3.17", "ninja"]
+build-backend = "setuptools.build_meta"
```
[the remark this answers is not in the document]
Comment (henryiii) [reply pyproject.toml:3]: The default is `build-backend = "setuptools.build_meta:__legacy__"`, which has compatibility hacks to try to support a package that never expected to be build in isolation. Though it's still a different procedure, so it tends to work on both or fail on both. The main user-facing difference that I often see is the normal backend doesn't automatically inject the local directory into sys.path, so you can't import locally in `setup.py` without manually doing that. The reasoning was that importing a local package in `setup.py` was an antipattern and should not be encouraged.

**Pull request merged**